Readme md analytics

juanmaguitar opened the pull request:
**What**:

Adding README.md's of analytic packages and templates to create similar README.md's for the rest of the packages

**Why**:

So every package has the proper info in NPM

**How**:

Creating the README.md's using on each folder

```
npx readme-md-generator -p ../../.github/README_TEMPLATE/package-readme.md -y
```

And then adding custom info on each README.md

**Tasks**:

<!-- Changesets are necessary if your changes should release any packages.
Run `npx changeset` to create a changeset.
More info at https://docs.frontity.org/contributing/code-contribution-guide#what-is-a-changeset -->

**Unrelated Tasks**

- Code
- TSDocs
- TypeScript
- Unit tests
- End to end tests
- TypeScript tests
- Update starter themes
- Update other packages
- Link to PR in [Documentation](https://github.com/frontity/gitbook-docs/)
- Community discussions
- Changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)

**Additional Comments**

<!-- Feel free to add any additional comments. -->

juanmaguitar committed:
Added README

juanmaguitar committed:
README md's full info

juanmaguitar committed:
added CHANGELOG

juanmaguitar committed:
added changset so the README are published to NPM

luisherranz commented on packages/google-analytics/README.md:
```diff
@@ -1,10 +1,67 @@
+      name: "@frontity/google-analytics",
+      state: {
+        googleAnalytics: {
+          trackingIds: ['UA-12345678-9']
```
For the basic usage I would show the singular option `trackingId`:
- https://github.com/frontity/frontity/blob/dev/packages/google-analytics/types.ts#L30

The plural `trackingIds` is only needed for advanced sites that need more than one tracking ID.

juanmaguitar replied:
Done! 👍

luisherranz commented on packages/google-tag-manager/README.md:
```diff
@@ -1,13 +1,67 @@
+      name: "@frontity/google-tag-manager",
+      state: {
+        googleTagManager: {
+          containerIds: ['GTM-BCDFGHJ']
```
For the basic usage I would show the singular option `containerId`:

- https://github.com/frontity/frontity/blob/dev/packages/google-tag-manager/types.ts#L19

The plural `containerIds` is only needed for advanced sites that need more than one container ID.

juanmaguitar replied:
Done! 👍

luisherranz commented on packages/google-tag-manager/README.md:
```diff
@@ -1,13 +1,67 @@
-> ## Beta version
->
-> Please, bear in mind that this package is currently in beta version. It hasn't proper tests or docs and its implementation may change with the final release.
```
We still need to maintain a warning about the beta version.

juanmaguitar replied:
Done! 👍

luisherranz commented on packages/google-analytics/README.md:
```diff
@@ -1,10 +1,67 @@
-> ## Beta version
-> Please, bear in mind that this package is currently in beta version. It hasn't proper tests or docs and its implementation may change with the final release.
```
We still need to maintain a warning about the beta version.

juanmaguitar replied:
Done! 👍

luisherranz reviewed (changes requested):
They are great, thanks @juanma 👏👏

BTW, what is your idea for the Usage section? Add a basic Usage section here and refer to the docs for complete information?

juanmaguitar committed:
better examples & warnings beta

SantosGuillamot commented:
I really like them, thanks @juanmaguitar ! :slightly_smiling_face: I'm just missing some links to the docs, but I guess it's because we still don't have that pages created.

Btw, quick question, if we change something in the footer template we still need to update the packages manually right?

juanmaguitar commented:
> BTW, what is your idea for the Usage section? Add a basic Usage section here and refer to the docs for complete information?

Yes! Adding some basic example of how to use the package and then adding a link to the docs to more advanced usage

juanmaguitar commented:
> I really like them, thanks @juanmaguitar ! 🙂 I'm just missing some links to the docs, but I guess it's because we still don't have that pages created.

Yes, these packages doesn't have specific documentation in the docs. Those whoe have it will have the proper link to the docs

> Btw, quick question, if we change something in the footer template we still need to update the packages manually right?

You're right. The templates are used to create new `README.md`. 
Any update on a `README.md` has to be done manually or re-creating a new README.md using the templates and then updating it

luisherranz reviewed (approved):
Thanks @juanmaguitar  👍

mburridge committed:
Reworded Beta warning message

mburridge committed:
Reworded Beta warning message

mburridge committed:
Updates to footer template

Minor changes, and also made the resources consistent with https://docs.frontity.org/about/get-involved .

juanmaguitar committed:
added link to docs

juanmaguitar commented:
@luisherranz Is this ready to be merged?

luisherranz committed:
Merge branch 'dev' into README-md-analytics

luisherranz commented:
Sure. It's approved 😄

_(I've rerun the tests because an e2e test failed. We'll have to check that.)_
